rtt_rosparam/src/rtt_rosparam_service.cpp

```diff
@@ -12,6 +12,29 @@
 
 #include <ros/ros.h>
 
+#ifndef DECLARE_ROSPARAM_GETTER_SETTER
+#define DECLARE_ROSPARAM_GETTER_SETTER(function_name,return_type) \
```

Comment:
Something like this should work (untested):
```cpp

#ifndef ADD_ROSPARAM_OPERATION
#define ADD_ROSPARAM_OPERATION(function_name, return_type) \
  this->addOperation("get"#function_name, &ROSParamService::getParamImpl< return_type >, this).doc("Get a " #return_type " from ROS parameter server"); \
  this->addOperation("set"#function_name, &ROSParamService::setParamImpl< return_type >, this).doc("Set a " #return_type " in the ROS parameter server");
#endif

class ROSParamService : public RTT::Service
{
public:
  ROSParamService(RTT::TaskContext *owner)
    : RTT::Service("rosparam", owner)
  {
    // [...]

    ADD_ROSPARAM_OPERATION(String, std::string)
    ADD_ROSPARAM_OPERATION(Double, double)
    ADD_ROSPARAM_OPERATION(Float, float)
    ADD_ROSPARAM_OPERATION(Int, int)
    ADD_ROSPARAM_OPERATION(Bool, bool)

    // Vector parameters
    ADD_ROSPARAM_OPERATION(VectorOfString, std::vector<string>)
    ADD_ROSPARAM_OPERATION(VectorOfDouble, std::vector<double>)
  }

  // [...]

private:
  template <typename T> bool getParamImpl(const std::string& ros_param_name, T& value)
  {
      if (!ros::param::get(ros_param_name, value)) {
          RTT::log(RTT::Debug) << "ROS Parameter \"" << ros_param_name << "\" not found on the parameter server!" << RTT::endlog();
          return false;
      }
      return true;
  }

  template <typename T> void setParamImpl(const std::string& ros_param_name, const T& value)
  {
      ros::param::set(ros_param_name, value);
  }
};
```

Note that I also updated the debug log message to include the return type instead of the function name and change the name of the macro to `ADD_ROSPARAM_OPERATION`. The private method `setParamImpl()` could be omitted if `&ros::param::set` is bound directly for the operation , but I like the explicit variant more.